borg/hashindex.pyx

```diff
@@ -63,7 +77,9 @@ cdef class IndexBase:
             path = os.fsencode(path)
             self.index = hashindex_read(path)
             if not self.index:
-                raise Exception('hashindex_read failed')
+                if errno:
+                    raise OSError(errno, decoded_strerror(errno), os.fsdecode(path))
```

Comment:
hmm, is this going to get us into trouble if there is some crap in path or errno msg, that does not output to the console?